File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    dic = { 'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 
           'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9' }
    
    for dic_key, dic_val in zip(dic, dic.values()):
        if s.isdigit():
            break
        s = s.replace(dic_key, dic_val)
        
    answer = int(s)
    
    return answer

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        String[] arr = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"};
        
        for(int i=0; i<arr.length; i++) 
            s = s.replace(arr[i], String.valueOf(i));
        answer = Integer.parseInt(s);
        
        return answer;
    }
}